public/api-v1.php

```diff
@@ -412,6 +413,81 @@
     return $response;
 });
 
+/*********************************
+* POST /firmware
+**********************************/
+$app->post('/firmware', function(Request $request, Response $response) use ($app) {
+    $uploadedFile = array_pop($request->getUploadedFiles());
+    if ($uploadedFile->getError() === UPLOAD_ERR_OK) {
+        if (! preg_match('/^[a-zA-Z0-9\-_\.()]+$/', $uploadedFile->getClientFilename())) {
```

Comment:
Do we need to escape brakets here?

```suggestion
        if (! preg_match('/^[a-zA-Z0-9\-_\.\(\)]+$/', $uploadedFile->getClientFilename())) {
```